test/test_gather.cpp

```diff
@@ -555,4 +584,683 @@ TEST_F(IndexingOpTest, GatherBroadcastInput_CUDA) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100, 100});
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = makeSymbolicTensor(1, DataType::Int);
+  fusion.addInput(tv1);
+
+  auto tv2 = add(tv0, IrBuilder::create<Double>(1));
+  auto tv3 = broadcast(tv1, {false, true});
+  auto tv4 = take_along_axis(tv2, tv3, 1);
+  auto tv5 = squeeze(tv4, std::vector<bool>{false, true});
+  auto tv6 = sum(tv5, {0});
+  fusion.addOutput(tv6);
+
+  at::manual_seed(0);
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto options_i = at::TensorOptions().dtype(at::kLong).device(at::kCUDA, 0);
+  auto t0 = at::randn(shape, options);
+  auto t1 = at::randint(0, shape[1], {shape[0]}, options_i);
+  std::vector<c10::IValue> aten_inputs = {t0, t1};
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
+  auto outputs = fec.runFusionWithInputs(aten_inputs);
+
+  validateSegmentation(
+      fec.getMostRecentKernelRuntime(),
+      {ScheduleHeuristic::PointWise, ScheduleHeuristic::Reduction});
+
+  auto t4 = at::take_along_dim(t0.to(at::kDouble) + 1, t1.unsqueeze(-1), 1);
+  auto ref = t4.squeeze(1).sum({0});
+
+  testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
+}
+
+// take_along_axis then reduction. Should not be segmented.
+TEST_F(IndexingOpTest, TakeAlongAxisIntermediateTensorReduction3_CUDA) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape_before_gather({100, 100});
+  std::vector<int64_t> shape_after_gather({shape_before_gather[0], 120});
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = makeSymbolicTensor(2, DataType::Int);
+  fusion.addInput(tv1);
+
+  auto tv2 = add(tv0, IrBuilder::create<Double>(1));
+  auto tv3 = take_along_axis(tv2, tv1, 1);
+  auto tv4 = sum(tv3, {1});
+  fusion.addOutput(tv4);
+
+  at::manual_seed(0);
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto options_i = at::TensorOptions().dtype(at::kLong).device(at::kCUDA, 0);
+  auto t0 = at::randn(shape_before_gather, options);
+  auto t1 =
+      at::randint(0, shape_before_gather[1], shape_after_gather, options_i);
+  std::vector<c10::IValue> aten_inputs = {t0, t1};
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
+  auto outputs = fec.runFusionWithInputs(aten_inputs);
+
+  validateSegmentation(
+      fec.getMostRecentKernelRuntime(), {ScheduleHeuristic::Reduction});
+
+  auto ref = at::take_along_dim(t0.to(at::kDouble) + 1, t1, 1).sum({1});
+
+  testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
+}
+
+// Similar to TakeAlongAxisIntermediateTensorReduction2, but no
+// squeeze of the consumer ID of the indexed domain. Should not be segmented.
+TEST_F(IndexingOpTest, TakeAlongAxisIntermediateTensorReduction4_CUDA) {
```

Comment:
I am seeing this error with this test on my 3090: 

```
[ RUN      ] IndexingOpTest.TakeAlongAxisIntermediateTensorReduction4_CUDA
error: CUDA_ERROR_COOPERATIVE_LAUNCH_TOO_LARGE failed with error too many blocks in cooperative launch
```

Reply:
Oops, sorry. I'll update the test.

Reply:
This should fix the error https://github.com/NVIDIA/Fuser/pull/292. Only 40 blocks are launched.

Reply:
Cross referencing https://github.com/NVIDIA/Fuser/issues/284